src/egon/data/datasets/electrical_neighbours.py

```diff
@@ -1267,13 +1273,391 @@ def tyndp_demand():
+    return df
+
+def entsoe_historic_demand(entsoe_token=None, year_start="20190101", year_end="20200101"):
+    client = entsoe.EntsoePandasClient(api_key=entsoe_token)
+    
+    start = pd.Timestamp(year_start, tz="Europe/Brussels")
+    end = pd.Timestamp(year_end, tz="Europe/Brussels")
+    start_gb = start.tz_convert("Europe/London")
+    end_gb = end.tz_convert("Europe/London")
+
+    countries= ["LU", "AT", "FR", "NL", 
+                "DK_1", "DK_2", "PL", "CH", "NO", "BE", "SE", "GB"]
+    
+    
+
+    # todo: define wanted countries
+
+
+    not_retrieved = []
+    dfs = []
+
+    for country in countries:
+        if country == 'GB':
+            kwargs = dict(start=start_gb, end=end_gb)
+        else:
+            kwargs = dict(start=start, end=end)
+        try:
+            country_data = client.query_load(country, **kwargs).resample("H")["Actual Load"].mean()
+            if country == 'GB':
+                country_data.index = country_data.index.tz_convert("Europe/Brussels")
+            dfs.append(country_data)
+        except (entsoe.exceptions.NoMatchingDataError, requests.HTTPError):
+            not_retrieved.append(country)
+            pass
+    if not_retrieved:
+        logger.warning(
+            f"Data for country (-ies) {', '.join(not_retrieved)} could not be retrieved."
+        )
+
+    df = pd.concat(dfs, axis=1)
+    df.columns = countries
+    df.index = pd.date_range(year_start, periods=8760 , freq="H")
+
+    return df
+
+def map_carriers_entsoe():
+    """Map carriers from entsoe-data to carriers used in eGon
+    Returns
+    -------
+    dict
+        Carrier from entsoe to eGon
+    """
+    return {
+            'Biomass': "biomass",
+            'Fossil Brown coal/Lignite': "lignite",
+            'Fossil Coal-derived gas': "coal",
+            'Fossil Gas': "CH4_NG",
+            'Fossil Hard coal': "coal",
+            'Fossil Oil': "oil",
+            'Fossil Oil shale': "oil",
+            'Fossil Peat': "biomass",
+            'Geothermal': "geo_thermal",
+            'Hydro Pumped Storage': "Hydro Pumped Storage",
+            'Hydro Run-of-river and poundage': "run_of_river",
+            'Hydro Water Reservoir': "reservoir",
+            'Marine': "marine",
+            'Nuclear': "nuclear",
+            'Other': "others",
+            'Other renewable': "others",
+            'Solar': "solar",
+            'Waste': "others",
+            'Wind Offshore': "wind_offshore",
+            'Wind Onshore': "wind_onshore",
+        }
+
+def entsoe_to_bus_etrago():
+    map_entsoe= pd.Series({"LU": "LU00",
+                 "AT": "AT00",
+                 "FR": "FR00",
+                 "NL": "NL00",
+                 "DK_1": "DK00",
+                 "DK_2": "DKE1",
+                 "PL": "PL00",
+                 "CH": "CH00",
+                 "NO": "NO00",
+                 "BE": "BE00",
+                 "SE": "SE00",
+                 "GB": "UK00"})
+
+    for_bus = get_foreign_bus_id()
```

Comment:
this function leads to an error, when only executing the status2019 scenario since the corresponding function explicitly addresses the eGon2035 scenario...

Reply:
I tried to solve this problem by 35403d5dd88d2a58ce61da0f0e589df1f4c1fe69 . But it did not work and both status2010 entsoe tasks run into this problem:

```
 WARNING: No data returned by statement: 
 SELECT bus_id, ST_Buffer(geom, 1) as geom, country
        FROM grid.egon_etrago_bus
        WHERE scn_name = '{scenario}'
        AND carrier = 'AC'
        AND v_nom = 380.
        AND country != 'DE'
        AND bus_id NOT IN (
            SELECT bus_i
            FROM osmtgmod_results.bus_data)
[2023-05-23 14:31:52,583] {taskinstance.py:1150} ERROR - index 0 is out of bounds for axis 0 with size 0
Traceback (most recent call last):
  File "/home/powerd/sh-powerd-data/venv/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 984, in _run_raw_task
    result = task_copy.execute(context=context)
  File "/home/powerd/sh-powerd-data/powerd-data/src/egon/data/datasets/__init__.py", line 200, in skip_task
    result = super(type(task), task).execute(*xs, **ks)
  File "/home/powerd/sh-powerd-data/venv/lib/python3.8/site-packages/airflow/operators/python_operator.py", line 113, in execute
    return_value = self.execute_callable()
  File "/home/powerd/sh-powerd-data/venv/lib/python3.8/site-packages/airflow/operators/python_operator.py", line 118, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/home/powerd/sh-powerd-data/powerd-data/src/egon/data/datasets/electrical_neighbours.py", line 1457, in insert_generators_sq
    entsoe_to_bus = entsoe_to_bus_etrago()
  File "/home/powerd/sh-powerd-data/powerd-data/src/egon/data/datasets/electrical_neighbours.py", line 1403, in entsoe_to_bus_etrago
    for_bus = get_foreign_bus_id(scenario='status2019')
  File "/home/powerd/sh-powerd-data/powerd-data/src/egon/data/datasets/electrical_neighbours.py", line 780, in get_foreign_bus_id
    buses.loc[i, "bus_id"] = distance[
IndexError: index 0 is out of bounds for axis 0 with size 0

@CarlosEpia any idea how to fix this?
```

Reply:
I did the mentioned changed and the tasks worked nicely in the server for SH.

Reply:
Thanks @CarlosEpia !